src/Metadata/Resource/ResourceMetadataCollection.php

```diff
@@ -80,6 +80,30 @@ public function getOperation(?string $operationName = null, bool $forceCollectio
         $this->handleNotFound($operationName, $metadata);
     }
 
+    /**
+     * Find a Get operation which matches the uriTemplate.
+     */
+    public function matchOperation(string $uriTemplate): Operation
```

Comment:
What about avoiding this new method and instead:

- when `itemUriTemplate` is given, store an `itemOperationName`
- then use `$resourceCollection->getOperation($operation->getItemOperationName())` 

If needed, a user can specify `itemOperationName` instead of the template.